p2p/switch.go

```diff
@@ -279,20 +281,32 @@ func (sw *Switch) Broadcast(chID byte, msgBytes []byte) chan bool {
 	return successChan
 }
 
-// NumPeers returns the count of outbound/inbound and outbound-dialing peers.
-func (sw *Switch) NumPeers() (outbound, inbound, dialing int) {
+// NumPeers returns the count of outbound/inbound, outbound-dialing and unconditional peers.
+func (sw *Switch) NumPeers() (outbound, inbound, dialing, unconditionalOutbound, unconditionalInbound int) {
 	peers := sw.peers.List()
 	for _, peer := range peers {
+		unconditional := sw.IsUnconditionalPeer(peer.ID())
 		if peer.IsOutbound() {
 			outbound++
+			if unconditional {
+				unconditionalOutbound++
+			}
 		} else {
 			inbound++
+			if unconditional {
+				unconditionalInbound++
+			}
 		}
 	}
 	dialing = sw.dialing.Size()
 	return
 }
 
+func (sw *Switch) IsUnconditionalPeer(id ID) bool {
```

Comment:
```suggestion
func (sw *Switch) IsPeerUnconditional(id ID) bool {
```

to be consistent with IsPeerPersistent (i.e. Is + Peer + QualityName)